packages/labs/src/multi-select/multiSelect.tsx

```diff
@@ -0,0 +1,299 @@
+    HTMLInputProps,
+    IPopoverProps,
+    Keys,
+    Popover,
+    Position,
+    Utils,
+} from "@blueprintjs/core";
+import { IListItemsProps, IQueryListRendererProps, QueryList } from "../query-list/queryList";
+import { ISelectItemRendererProps } from "../select/select";
+import { ITagInputProps, TagInput } from "../tag-input/tagInput";
+
+export interface IMultiSelectProps<T> extends IListItemsProps<T> {
+    /**
+     * Custom renderer for an item in the dropdown list. Receives a boolean indicating whether
+     * this item is active (selected by keyboard arrows) and an `onClick` event handler that
+     * should be attached to the returned element.
+     */
+    itemRenderer: (itemProps: IMultiSelectItemRendererProps<T>) => JSX.Element;
+
+    /** React child to render when filtering items returns zero results. */
+    noResults?: string | JSX.Element;
+
+    /**
+     * Whether the popover opens on key down or when `TagInput` is focused.
+     * @default false
+     */
+    openOnKeyDown?: boolean;
+
+    /** Props to spread to `Popover`. Note that `content` cannot be changed. */
+    popoverProps?: Partial<IPopoverProps>;
+
+    /**
+     * Whether the filtering state should be reset to initial when an item is selected
+     * (immediately before `onItemSelect` is invoked), or when the popover closes.
+     * The query will become the empty string and the first item will be made active.
+     * @default true
+     */
+    resetOnSelect?: boolean;
+
+    /** Controlled selected values. */
+    selectedItems?: T[];
+
+    /** Props to spread to `TagInput`. */
+    tagInputProps?: Partial<ITagInputProps>;
+
+    /** Custom renderer to transform an item into a string for tags. */
+    tagRenderer: (item: T) => string;
+}
+
+export interface IMultiSelectItemRendererProps<T> extends ISelectItemRendererProps<T> {
+    /**
+     * Whether the item is selected.
+     */
+    isSelected?: boolean;
+}
+
+export interface IMultiSelectState<T> {
+    activeItem?: T;
+    isOpen?: boolean;
+    query?: string;
+    selectedItems?: T[];
+}
+
+@PureRender
+export class MultiSelect<T> extends AbstractComponent<IMultiSelectProps<T>, IMultiSelectState<T>> {
+    public static displayName = "Blueprint.MultiSelect";
+
+    public static ofType<T>() {
+        return MultiSelect as new () => MultiSelect<T>;
+    }
+
+    public state: IMultiSelectState<T> = {
+        isOpen: false,
+        query: "",
+        selectedItems: [],
+    };
+
+    private TypedQueryList = QueryList.ofType<T>();
+    private input: HTMLInputElement;
+    private queryList: QueryList<T>;
+    private queryListKeyDown: React.EventHandler<React.KeyboardEvent<HTMLElement>>;
+    private refHandlers = {
+        input: (ref: HTMLInputElement) => this.input = ref,
+        queryList: (ref: QueryList<T>) => this.queryList = ref,
+    };
+
+    public componentDidMount() {
+        // can't use defaultProps because "static members
+        // cannot reference class type parameters"
```

Comment:
That's the problem I'm hitting:
![image](https://user-images.githubusercontent.com/199754/27563452-9a5b32cc-5a86-11e7-8b1b-127ba47e57f9.png)

I'd still like to be able to set default props for `openOnKeyDown` and `resetOnSelect`